internal/clients/team4/speaker.go

```diff
@@ -11,12 +11,49 @@ type speaker struct {
 	parent *client
 }
 
-//s.parent.ServerReadHandle.GetGameConfig()
+//SpeakerActionPriorities indicate speaker actions in order of priority
+var SpeakerActionPriorities = []string{
+	"SetVotingResult",
+	"SetRuleToVote",
+	"AnnounceVotingResult",
+	"UpdateRules",
+	"AppointNextJudge",
+}
+
+func (s *speaker) getSpeakerBudget() shared.Resources {
+	return s.parent.ServerReadHandle.GetGameState().IIGORolesBudget[shared.Speaker]
+}
+
+func (s *speaker) getActionsCost(actions []string) shared.Resources {
+	gameconfig := s.parent.ServerReadHandle.GetGameConfig().IIGOClientConfig
+	costs := map[string]shared.Resources{
+		"SetVotingResult":      gameconfig.SetVotingResultActionCost,
+		"SetRuleToVote":        gameconfig.SetRuleToVoteActionCost,
+		"AnnounceVotingResult": gameconfig.AnnounceVotingResultActionCost,
+		"UpdateRules":          gameconfig.UpdateRulesActionCost,
+		"AppointNextJudge":     gameconfig.AppointNextJudgeActionCost,
+	}
+	var SumOfCosts shared.Resources = 0
+	for _, action := range actions {
+		SumOfCosts += costs[action]
+	}
+	return SumOfCosts
+}
+
+func (s *speaker) getHigherPriorityActionsCost(baseaction string) shared.Resources {
+	actionindex := len(SpeakerActionPriorities)
+	for i, action := range SpeakerActionPriorities {
+		if action == baseaction {
+			actionindex = i
+		}
+	}
+	return s.getActionsCost(SpeakerActionPriorities[:actionindex])
+}
 
 // PayJudge is used for paying judge for his service
-/*func (s *speaker) PayJudge() shared.SpeakerReturnContent {
-	JudgeSalaryRule, ok := rules.RulesInPlay["salary_cycle_judge"]
+func (s *speaker) PayJudge() shared.SpeakerReturnContent {
 	var JudgeSalary shared.Resources = 0
+	JudgeSalaryRule, ok := rules.RulesInPlay["salary_cycle_judge"]
 	if ok {
 		JudgeSalary = shared.Resources(JudgeSalaryRule.ApplicableMatrix.At(0, 1))
```

Comment:
Just a question about this rule, is the rule's aux value always going to be "0"? (i.e. is it always comparing (-1)*judgeSalary + constant == 0 ?)

Reply:
This value is taken from the baseclient, so I believe it's fine

Reply:
The salary cycle rules are there just for the sake of the agents being able to change the value. None of the salary cycle rules are actually logged and monitored. We made them into rules because they're the only 'global custom variable' option we had. (for more insight on this decision please ask @Spruudis )

Reply:
@MikeIsMike you are right. By changing the constant the agents can change the expected salary. @777jameskim the rule is logged and monitoring should catch an agent breaking this rule.